### DISCUSSION TITLE
How to simulate both UAVs and cars using AirSim?

### DISCUSSION THREAD

**Discussion by UrYuWang** [microsoft/AirSim#3372] [Support Q&A] [answered]
Hello.
I want to use AirSim to simulate UAV and car at the same time. I found there is one solution for Linux, and I wonder if there is a solution for Windows 10.
Thanks in advance.

**Comment by jonyMarino** [comment]
Hi @UrYuWang! and welcome! can you share that solution for Linux, please?

**Reply by ahmed-elsaharti** [reply]
I think @UrYuWang is referring to [the repo](https://github.com/IdoMatan/AirsimAugmentedFiles) used in [this guide](https://medium.com/@idoglanz/setting-up-microsoft-airsim-to-simulate-a-drone-and-car-together-708079b2d0f). By the looks of it though, it seems like this should also work for Windows too.

**Comment by UrYuWang** [comment]
There is one called AirSim augmentation. You can search it in GitHub.

Jonathan <notifications@github.com> 于 2021年2月9日周二 11:42写道：

**Comment by UrYuWang** [comment]
I couldn't run the bash files. That is the issue of using it on windows.

Ahmed Elsaharti <notifications@github.com> 于 2021年2月9日周二 下午1:45写道：

**Comment by ahmed-elsaharti** [comment, marked as answer]
Hi @UrYuWang, I haven't tested this myself yet, but based on what you're saying I think the issue is that after adding the modified files you're following the [Linux build steps](https://microsoft.github.io/AirSim/build_linux/) on windows rather than the [Windows build steps](https://microsoft.github.io/AirSim/build_windows). Try [building this way](https://microsoft.github.io/AirSim/build_windows/#build-airsim) after adding the files.

**Reply by UrYuWang** [reply]
After I run the build.cmd file and replace the plugins folder, it pops "invalid mode: Both". I don't know if this works for you.
There must be something wrong I think.

**Reply by ahmed-elsaharti** [reply]
As [the guide](https://medium.com/@idoglanz/setting-up-microsoft-airsim-to-simulate-a-drone-and-car-together-708079b2d0f) suggests, you need to replace the files first before building. 
That being said, I dont think this mod is directly compatible with the latest AirSim commit due to multiple conflicts (based on a test I just ran, there was a decent amount of errors on trying to compile)
If this were to work, I think you would need to go through all the files in the [custom repo](https://github.com/IdoMatan/AirsimAugmentedFiles) and copy the effective changes to the latest AirSim commit.
OR
You could try to go back to an AirSim commit from 6 months ago (around the time the custom repo was made) and run that with the custom files rather than the latest commit.

**Comment by UrYuWang** [comment]
Hi, Ahmed.
Thanks a lot! I think you pointed out the issue directly!

Best,
Yu

Ahmed Elsaharti <notifications@github.com> 于2021年2月9日周二 下午2:07写道：

> Hi @UrYuWang <https://github.com/UrYuWang>, I haven't tested this myself
> yet, but based on what you're saying I think the issue is that after adding
> the modified files you're following the Linux build steps
> <https://microsoft.github.io/AirSim/build_linux/> on windows rather than
> the Windows build steps <https://microsoft.github.io/AirSim/build_windows>.
> Try building this way
> <https://microsoft.github.io/AirSim/build_windows/#build-airsim> after
> adding the files.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/AirSim/discussions/3372#discussioncomment-354171>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGP4WDNYXEZ63ZYRGXKT54TS6GBVTANCNFSM4XDTH4GA>
> .
>


-- 
Have a good day.
Yu

**Comment by jonyMarino** [comment]
Thanks, Ahmed!
@UrYuWang, can you share with us why you are interested in simulating cars and drones together?

**Reply by UrYuWang** [reply]
I am working on a car chasing simulation. 
I am still facing the issue on windows. It shows "both" is not a valid simulation type.
I literally don't know which part went wrong. Have you tried it on windows?

**Comment by UrYuWang** [comment]
Thanks for the tip!

Ahmed Elsaharti <notifications@github.com> 于2021年2月11日周四 下午8:09写道：

> As the guide
> <https://medium.com/@idoglanz/setting-up-microsoft-airsim-to-simulate-a-drone-and-car-together-708079b2d0f>
> suggests, you need to replace the files first before building.
> That being said, I dont think this mod is directly compatible with the
> latest AirSim commit due to multiple conflicts (based on a test I just ran,
> there was a decent amount of errors on trying to compile)
> If this were to work, I think you would need to go through all the files
> in the custom repo <https://github.com/IdoMatan/AirsimAugmentedFiles> and
> copy the effective changes to the latest AirSim commit.
> OR
> You could try to go back to an AirSim commit from 6 months ago (around the
> time the custom repo was made) and run that with the custom files rather
> than the latest commit.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/AirSim/discussions/3372#discussioncomment-361446>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGP4WDKURJVYSPSPIKJPD3DS6R5WLANCNFSM4XDTH4GA>
> .
>


-- 
Have a good day.
Yu
